hail/python/hailtop/config/user_config.py

```diff
@@ -0,0 +1,20 @@
+import os
+import configparser
+
+user_config = None
+
+
+def get_user_config_path():
+    config_file = os.environ.get('XDG_CONFIG_HOME', os.path.expanduser('~/.config')) + '/hail/config.yaml'
+    os.makedirs(os.path.dirname(config_file), exist_ok=True)
```

Comment:
This is a bit weird to me that you're creating directories here.

Reply:
Agreed, I moved it to get_user_config, which now unconditionally ensures the directories *and* the file exist.